[quant][graphmode] Move avg_pool/adaptive_avg_pool to general value ops

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38278 [quant][graphmode] Support ops with fixed quantization parameters
* #38277 [quant][graphmode][refactor] insertDeQuantForAllUse
* #38276 [quant][graphmode][refactor] Refactor propagateQuantizationOps
* #38166 [quant][graphmode] Move leaky_relu to general value op map
* #38165 [quant][graphmode] Move elu to general value ops map
* #38164 [quant][graphmode] Move hardtanh to general value ops map
* #38163 [quant][graphmode] Move clamp to general value ops map
* #38162 [quant][graphmode] Move interpolate to general value ops
* #38161 [quant][graphmode][refactor] Factor out common parts of general value ops
* #38160 [quant][graphmode] Move `aten::mean` to general value ops
* **#38159 [quant][graphmode] Move avg_pool/adaptive_avg_pool to general value ops**

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_quantize_general_value_ops
Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512409](https://our.internmc.facebook.com/intern/diff/D21512409)

jerryzh168 committed:
[quant][graphmode] Move avg_pool/adaptive_avg_pool to general value ops

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_quantize_general_value_ops
Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Move avg_pool/adaptive_avg_pool to general value ops"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_quantize_general_value_ops
Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1672,17 +1669,15 @@ def forward(self, x):
-        general_value_op_quant = 2
-        FileCheck().check_count("aten::quantize_per_tensor", conv_op_quant + general_value_op_quant + 1, exactly=True) \
+        general_value_op_quant = 12
+        FileCheck().check_count("aten::quantize_per_tensor(", conv_op_quant + general_value_op_quant + 1, exactly=True) \
```
How does the IR look here? Is it conv->q->dq->avg_pool->q->dq->avg_pool-.>?

jerryzh168 replied:
yes

supriyar commented on test/quantization/test_quantize_script.py:
```diff
@@ -1672,17 +1669,15 @@ def forward(self, x):
-        m = wrap_cpp_module(torch._C._jit_pass_insert_observers(
-            torch.jit.script(M())._c, 'forward', {'': qconfig}, inplace=False))
         m2 = convert_script(m, debug=False)
         FileCheck().check_count("aten::quantize_per_tensor(", 1, exactly=True) \
```
Can we also check for presence of the pooling ops in the final graph?

jerryzh168 replied:
that is probably not worth it since that would mean spelling out the graph. for example, in the later PR, we have some more complicated ops like interpolate.
What we are checking here is the number of inserted quantize_per_tensor ops are correct, and I think it is enough to ensure the correctness

supriyar reviewed (approved):
LG, left a small comment

jerryzh168 committed:
Update on "[quant][graphmode] Move avg_pool/adaptive_avg_pool to general value ops"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_quantize_general_value_ops
Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512409](https://our.internmc.facebook.com/intern/diff/D21512409)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Move avg_pool/adaptive_avg_pool to general value ops"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_quantize_general_value_ops
Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21512409](https://our.internmc.facebook.com/intern/diff/D21512409)

[ghstack-poisoned]

jerryzh168 commented:
closing this due to some ghstack issue, created a new one: https://github.com/pytorch/pytorch/pull/38330
